presto-phoenix/src/main/java/io/prestosql/plugin/phoenix/QueryBuilder.java

```diff
@@ -0,0 +1,364 @@
+            String columnNames = decomposePrimaryKeyColumn(desiredPCols).stream().map(PhoenixColumnHandle::getColumnName).collect(joining(", "));
+            sql.append(columnNames);
+        }
+        else {
+            String columnNames = decomposePrimaryKeyColumn(columns).stream().map(PhoenixColumnHandle::getColumnName).collect(joining(", "));
+            sql.append(columnNames);
+            if (columns.isEmpty()) {
+                sql.append("null");
+            }
+        }
+        sql.append(" FROM ");
+        if (!isNullOrEmpty(catalog)) {
+            sql.append(catalog).append('.');
+        }
+        if (!isNullOrEmpty(schema)) {
+            sql.append(schema).append('.');
+        }
+        sql.append(table);
+
+        List<TypeAndValue> accumulator = new ArrayList<>();
+
+        List<String> clauses = toConjuncts(columns, tupleDomain, accumulator);
+        if (!clauses.isEmpty()) {
+            sql.append(" WHERE ").append(Joiner.on(" AND ").join(clauses));
+        }
+
+        try (PhoenixPreparedStatement statement = connection.prepareStatement(sql.toString()).unwrap(PhoenixPreparedStatement.class)) {
+            for (int i = 0; i < accumulator.size(); i++) {
+                TypeAndValue typeAndValue = accumulator.get(i);
+                Type type = typeAndValue.getType();
+                if (type.equals(BigintType.BIGINT)) {
+                    statement.setLong(i + 1, (long) typeAndValue.getValue());
+                }
+                else if (type.equals(IntegerType.INTEGER)) {
+                    statement.setInt(i + 1, ((Number) typeAndValue.getValue()).intValue());
+                }
+                else if (type.equals(SmallintType.SMALLINT)) {
+                    statement.setShort(i + 1, ((Number) typeAndValue.getValue()).shortValue());
+                }
+                else if (type.equals(TinyintType.TINYINT)) {
+                    statement.setByte(i + 1, ((Number) typeAndValue.getValue()).byteValue());
+                }
+                else if (type.equals(DoubleType.DOUBLE)) {
+                    statement.setDouble(i + 1, (double) typeAndValue.getValue());
+                }
+                else if (type.equals(RealType.REAL)) {
+                    statement.setFloat(i + 1, intBitsToFloat(((Number) typeAndValue.getValue()).intValue()));
+                }
+                else if (type.equals(BooleanType.BOOLEAN)) {
+                    statement.setBoolean(i + 1, (boolean) typeAndValue.getValue());
+                }
+                else if (type.equals(DateType.DATE)) {
+                    long millis = DAYS.toMillis((long) typeAndValue.getValue());
+                    statement.setDate(i + 1, new Date(UTC.getMillisKeepLocal(DateTimeZone.getDefault(), millis)));
+                }
+                else if (type.equals(TimeType.TIME)) {
+                    statement.setTime(i + 1, new Time((long) typeAndValue.getValue()));
```

Comment:
this code should check `io.prestosql.spi.connector.ConnectorSession#isLegacyTimestamp`

This code is not correct  regardless of TIMESTAMP semantics
(unless JVM zone is UTC or equal to presto user session zone)